accumulo-store/src/main/java/gaffer/accumulostore/key/core/impl/classic/ClassicIteratorSettingsFactory.java

```diff
@@ -31,17 +32,26 @@
 
     @Override
     public IteratorSetting getEdgeEntityDirectionFilterIteratorSetting(final GetOperation<?, ?> operation) {
-        if (operation.getIncludeIncomingOutGoing() == IncludeIncomingOutgoingType.BOTH
-                && operation.getIncludeEdges() == IncludeEdgeType.ALL) {
+        final boolean includeEntities = operation.isIncludeEntities();
+        final IncludeEdgeType includeEdgeType = operation.getIncludeEdges();
+        final IncludeIncomingOutgoingType includeIncomingOutgoingType = operation.getIncludeIncomingOutGoing();
+        final boolean correctWayEdges = operation instanceof GetAllElements;
```

Comment:
This should be de-duplicate edges, also the setting should be set if the option from the accumulo constants  is set (you may want to de duplicate if you know what your doing)